aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -148,15 +148,15 @@ NestedTensorImpl::NestedTensorImpl(
     at::Tensor buffer,
     at::Tensor nested_size_tensor,
     at::Tensor nested_stride_tensor,
-    std::vector<int64_t> offsets)
+    std::vector<int64_t>&& offsets)
     : NestedTensorImpl(
           buffer.sizes()[0],
           buffer.storage(),
           generate_nested_key_set(buffer),
           buffer.dtype(),
           nested_size_tensor,
           nested_stride_tensor,
-          offsets) {
+          std::move(offsets)) {
```

Comment:
I do want to steal it and I want users to know its getting stolen hence make them copy when they call wrap_buffer or make them know that you can no longer own it buy callings std::move. IMO I think every nested tensor impl should own offsets since this is not a trivially copyable thing like a tensor.  

But maybe this is the wrong way of thinking about it